### DISCUSSION TITLE
How-To Automatically Starting YoutubeDL-Material?

### DISCUSSION THREAD

**Discussion by guillaumesoucy94** [Tzahi12345/YoutubeDL-Material#606] [Q&A]
Hello, 

How-to make YoutubeDL-Material to start automatically and also how-to running it in the background. 

I did try `nohup npm start &` 

but it give an error: 

`[1]+  Exit 1                  nohup npm start`

```
> backend@1.0.0 start /mnt/disk/vidconv.zsites.ca
> nodemon -q app.js

events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: EBADF: bad file descriptor, read
Emitted 'error' event at:
    at lazyFs.read (internal/fs/streams.js:165:12)
    at FSReqWrap.wrapper [as oncomplete] (fs.js:467:17)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! backend@1.0.0 start: `nodemon -q app.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the backend@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-05-05T13_08_03_289Z-debug.log
```

How I can make it work?

Regards, 

Guillaume

**Comment by Tzahi12345** [comment]
You can use pm2 which we use in the latest builds.

Something like: `pm2-runtime pm2.config.js`. In the repo we have a pm2.config.js file in the `backend` directory that could be helpful.

Using Docker will probably make things easier for you because it's easy to make containers launch on startup.
